de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/annotations/AnnotationManager.java

```diff
@@ -42,6 +41,11 @@
 
   private final Application application;
 
+  @Override
```

Comment:
Yes, theoretically this could cause concurrency problems when a new annotation is added while the current session is cleaned up (as the `AnnotationManager` is not threadsafe). But, as `stop()` is called before `dispose()` during the teardown of the session container, such an interaction should not be possible as all IntelliJ event handlers that could call it automatically will have been unregistered as part of the `stop()` call. Furthermore, as the AnnotationManager is only used by other session components (and the EditorManager, which treats it as a session component), it can't be called by other components at this stage as well.
So the AnnotationManager not being threadsafe should not matter here.

Executing the logic to remove the annotations from the editors should not be a problem either as the [logic removing the annotations](https://github.com/saros-project/saros/blob/be2a3422bcb36f484361e6928bfbed722756884b/intellij/src/saros/intellij/editor/annotations/AnnotationManager.java#L732) is synchronized on the AWT event dispatcher thread (see [l. 750](https://github.com/saros-project/saros/blob/be2a3422bcb36f484361e6928bfbed722756884b/intellij/src/saros/intellij/editor/annotations/AnnotationManager.java#L750)).